src/imitation/algorithms/dagger.py

```diff
@@ -372,7 +374,14 @@ def _load_all_demos(self) -> Tuple[types.Transitions, List[int]]:
         return demo_transitions, num_demos_by_round
 
     def _get_demo_paths(self, round_dir: pathlib.Path) -> List[pathlib.Path]:
-        return [round_dir / p for p in os.listdir(round_dir) if p.endswith(".npz")]
+        # listdir returns filenames in an arbitrary order that depends on the
+        # file system implementation:
+        # https://stackoverflow.com/questions/31534583/is-os-listdir-deterministic
+        # To ensure the order is consistent across file systems,
+        # we sort by the filename.
+        return [
```

Comment:
This list comprehension is getting a bit complicated (and now needs to be multiple lines). Consider splitting it up by using a temporary variable?